pkg/skaffold/runner/v1/new.go

```diff
@@ -82,8 +84,13 @@ func NewForConfig(runCtx *runcontext.RunContext) (*SkaffoldRunner, error) {
 		return nil, fmt.Errorf("creating tester: %w", err)
 	}
 	syncer := getSyncer(runCtx)
+
+	podSelectors := kubernetes.NewImageList()
+
+	logger := getLogger(runCtx, kubectlCLI, podSelectors)
```

Comment:
I like that idea a lot, I definitely planned on abstracting away the logging implementation from the `getLogger()` method, but your idea of a provider is much better than a big `switch` statement. I also want to move all of the `get*()` methods out of `new.go` (since this file is getting huge). WDYT about doing this in a follow up PR?

Reply:
sounds good!